eslint.config.mjs

```diff
@@ -0,0 +1,110 @@
+import { defineConfig, globalIgnores } from 'eslint/config'
+import typescriptEslint from '@typescript-eslint/eslint-plugin'
+import prettier from 'eslint-plugin-prettier'
+import preferArrow from 'eslint-plugin-prefer-arrow'
+import functional from 'eslint-plugin-functional'
+import globals from 'globals'
+import tsParser from '@typescript-eslint/parser'
+import path from 'node:path'
+import { fileURLToPath } from 'node:url'
+import js from '@eslint/js'
+import { FlatCompat } from '@eslint/eslintrc'
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: js.configs.recommended,
```

Comment:
I don't think we need the `FlatCompat`. It's more for plugis that haven't been upgraded to be compatible with the flat config.

The `js` can be added to `defineConfig` by just adding it directly to the array

Example:

https://typescript-eslint.io/getting-started#step-2-configuration

In that example, `js` is calles `eslint`

Reply:
Thanks for the heads up here, done!